### DISCUSSION TITLE
Statamic site in Plesk Obsidian (open_basedir restriction in effect)

### DISCUSSION THREAD

**Discussion by Sennik** [statamic/cms#5543] [Troubleshooting] [answered]
AH01071: Got error 'PHP message: PHP Fatal error: Uncaught ErrorException: file_exists(): open_basedir restriction in effect. File(/var/www/project/resources/views/errors/500.blade.php) is not within the allowed path(s): (/var/www/vhosts/project.com/:/tmp/) in /var/www/vhosts/project.com/httpdocs/vendor/laravel/framework/src/Illuminate/Filesystem/Filesystem.php:28

- First: it tries to load a blade file while Statamic uses Antlers (This is a fresh Statamic install which works on my DEV Ubuntu server). I never used Blade on this project
- The filepath misses 'vhosts/project.com/httpdocs', IMHO necessary because the file paths does not exists. I don't know why it makes this up because I did not configure this

I just uploaded the files & directories, just like I do on other projects. I also set the right permissions on the storage and bootstrap/cache directory. I don't understand why this is not working.

**Comment by afonic** [comment]
It tries to load the default error 500 page.

I'm not familiar with Plesk but you probably need to set the `public` of your project dir to be root of the vhost.

PS. The copy pasted error contains a link to your server, you might want to remove that.

**Reply by Sennik** [reply]
Never considered adding /public to the httpdocs root (/httpdocs/public), I always use a .htaccess file with a redirect to public/. It's not my first Laravel project ;-)

I tried to change the root to /httpdocs/public of course, same problem, so the problem is something else. Again, this exact project works like a charm on my dev server.

But thanks for the tip (Also for the tip about the link, not very smart ;-) too busy here)!

**Comment by duncanmcclean** [comment]
Anything in your logs file?

**Reply by Sennik** [reply]
Only the error which is shown...

**Comment by afonic** [comment, marked as answer]
It tries to show you the error 500 page and it can't access `/var/www/project/resources/views/errors/500.blade.php` to show it.

If `/var/www/project/` is your local dev dir it is probably cached somewhere, try clearing all caches, check your `.env` file (make sure it has a key) and probably try deleting the vendor folder and running composer again.

**Reply by Sennik** [reply]
Yes! You directed me to the solution! 

I did not have to delete the vendor folder, just `composer dump-autoload` was enough. 

One thing I had to change for that to work was the CLI PHP version. In Plesk you can install and select multiple PHP versions so every project can have it's own specific PHP version. But even if you choose PHP 8.1 in the project in Plesk's interface, the terminal PHP CLI version will not change. In my case this was defaulted to PHP 7.4 which was not recent enough of course.

But you can change that in the terminal: `export PATH=/opt/plesk/php/8.1/bin:$PATH;` This is temporary, after logging out it will be 7.4 again.

Thanks for the help y'all!

**Comment by nawidhaidari** [comment]
Hello All.
I have something similar issue. I am new to Plesk and do not know much more about PHP and Laravel and need some help.

when I check the Logs for my website. I have a PHP script with Laravel, the contact us page, the new user registration email, subscribers cant subscribe (Error "401: Your API key may be invalid, or you've attempted to access the wrong datacenter.") and the incoming and outgoing emails from the website do not work 

Here is the error message I get in the Log file:

AH01071: Got error 'PHP message: PHP Warning: file_exists(): open_basedir restriction in effect. File(/autoload.php) is not within the allowed path(s): (/var/www/vhosts/MySite.com/:/tmp/) in /var/www/vhosts/MySite.com/httpdocs/vendor/vonage/nexmo-bridge/src/Autoloader.php on line 69', referer: https://mysite.com/

I would appreciate any help and walking me through step by step to get this fixed. I am the Plesk Server Admin and the VPS Server in which I need to change anything in the root directory.

Thank you.
